File: InstitutionsService/InstitutionsService/RabbitMQ/Dto/InstitutionDto.cs
using InstitutionsService.Models;

namespace InstitutionsService.RabbitMQ.Dto
{
    /// <summary>
    /// Dto for replications. Institution model have more data that is needed for replication, so create smaller dto for replication.
    /// </summary>
    public class InstitutionDto
    {
        public long Id { get; set; }     
        public string Name { get; set; }
        public string RegCode { get; set; }
        public string? KMKR { get; set; }
        public Classifier? Type { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime? ValidTo { get; set; }
        public Address? Address { get; set; }
        public List<Translation>? Translations { get; set; }
    }
}

File: frontend/libs/services/src/lib/dto/institution-dto.ts
import { AddressDto } from "./address-dto";
import { ClassifierDto } from "./classifier-dto";
import { InstitutionReplicationDto } from "./institution-replication-dto";
import { TranslationDto } from "./translation-dto";

export interface InstitutionDto {
  id?: number;
  name?: string;
  name_translation_code?: string;
  reg_code?: string;
  kmkr?: string;
  address_id?: number;
  type_classifier_id?: number;
  valid_from?: Date;
  valid_to?: Date;
  translations?: TranslationDto[];
  type_classifier?: ClassifierDto;
  replications?: InstitutionReplicationDto[];
  address?: AddressDto,
}
